File: binary_search/binary_search.py
def insertion_sort(arr , value):
    left = 0 
    right = arr.size()-1
    mid = 0
    while left <=right:
        mid = (left + right)/2
        if(value < arr[mid]):
            right = mid - 1
        elif (value > arr[mid]):
            left = mid +1 
        else:
            return mid 
        return -1

    vetor = [ 1 , 2 , 3 , 4 , 5 , 6 , 7 ]
    print (insertion_sort(vetor , 3))

File: SomeAlgorithms/binary_search/binary_search.cpp
#include <iostream>
using namespace std;

int binary_search( int vetor[] , int value , int n){
    int left = 0 , right = n - 1;
    int mid;
    while(left  <= right ){
        mid = (left + right)/2;
        if(value > vetor[mid]){
            left = mid + 1;
        }
        else if ( value < vetor[mid]){
            right = mid - 1;
        }
        else{
            return mid;
        }
    }
    return -1;
  }

int main() {
    int n , number;
    scanf("%d", &n);
    scanf("%d", &number);
    int vetor[n];
    for(int i = 0 ; i<n ; i++){
        scanf("%d", &vetor[i]);
    }
    cout << binary_search(vetor , number , n );

  
}